src/visualization/types.jl

```diff
@@ -563,23 +567,38 @@ function PlotData1D(u, mesh, equations, solver, cache;
 end
 
 # Specializes the `PlotData1D` constructor for one-dimensional `DGMulti` solvers.
-function PlotData1D(u::StructArray, mesh, equations, dg::DGMulti{1}, cache;
+function PlotData1D(u, mesh, equations, dg::DGMulti{1}, cache;
                     solution_variables=nothing)
 
   solution_variables_ = digest_solution_variables(equations, solution_variables)
   variable_names = SVector(varnames(solution_variables_, equations))
 
   orientation_x = 0 # Set 'orientation' to zero on default.
 
-  data = map(x -> vcat(dg.basis.Vp * x, fill(NaN, 1, size(u, 2))),
-             StructArrays.components(solution_variables_.(u, equations)))
-  x_plot = vcat(dg.basis.Vp * mesh.md.x, fill(NaN, 1, size(u, 2)))
+  if u isa StructArray
+    data = map(x -> vcat(dg.basis.Vp * x, fill(NaN, 1, size(u, 2))),
+              StructArrays.components(solution_variables_.(u, equations)))
+    x = vcat(dg.basis.Vp * mesh.md.x, fill(NaN, 1, size(u, 2)))
+
+    # Here, we ensure that `DGMulti` visualization uses the same data layout and format
+    # as `TreeMesh`. This enables us to reuse existing plot recipes. In particular,
+    # `hcat(data...)` creates a matrix of size `num_plotting_points` by `nvariables(equations)`,
+    # with data on different elements separated by `NaNs`.
+    x_plot = vec(x)
+    data_plot = hcat(vec.(data)...)
+  else
+    data_tmp = dg.basis.Vp * solution_variables_.(u, equations)
```

Comment:
Can you add a comment noting what `solution_variables_` does?

Reply:
4d91b778